lib/moonshot/command_line.rb

```diff
@@ -0,0 +1,108 @@
+require 'thor'
+
+module Moonshot
+  # This class implements the command-line `moonshot` tool.
+  class CommandLine
+    # TODO: No reason this is class-level, make this a for-real class that needs to be initialized.
+    def self.run!
+      # Find the Moonfile in this project.
+      orig_dir = Dir.pwd
+
+      loop do
+        break if File.exist?('Moonfile')
+
+        if Dir.pwd == '/'
+          warn 'Could not find Moonfile!'
+          exit 1
+        else
+          Dir.chdir('..')
+        end
+      end
+
+      moonfile_dir = Dir.pwd
+      Dir.chdir(orig_dir)
+
+      # Load any plugins and CLI extensions relative to the Moonfile
+      if File.directory?(File.join(moonfile_dir, 'moonshot'))
+        load_cli_extensions(moonfile_dir)
+        load_plugins(moonfile_dir)
```

Comment:
Do you mean like one that is included via a gem? Requiring the gem, which would subclass Command/SSHCommand would load them in. I've got a branch for one of the internal shared extensions that I'm using locally, I'll send you a link.